src/systemtest/test_level2.py

```diff
@@ -918,148 +991,152 @@ def test_get_scan_v5(self, odinapi_service, fake_data):
             version=version,
             project=PROJECT_NAME,
         )
         if version == "v5":
             url = make_dev_url(url)
 
-        r = requests.get(url)
-        r.raise_for_status()
+        r = test_client.get(url, follow_redirects=True)
+        assert r.status_code == http.client.OK
 
         if version == "v4":
-            products = r.json()["Info"]["Products"]
+            products = r.json["Info"]["Products"]
         elif version == "v5":
-            products = r.json()["Data"]
+            products = r.json["Data"]
         else:
             raise NotImplementedError()
         assert len(products) == 3
 
     @pytest.mark.parametrize("version", ("v4", "v5"))
-    def test_get_products_for_freqmode(self, odinapi_service, fake_data, version):
+    def test_get_products_for_freqmode(self, test_client, fake_data, version):
         urlinfo, _ = fake_data
         url = make_dev_url(
             "{host}/rest_api/{version}/level2/{project}/{freqmode}/products/".format(
-                host=odinapi_service,
+                host="http://localhost",
                 version=version,
                 freqmode=urlinfo.freq_mode,
                 project=PROJECT_NAME,
             )
         )
-        r = requests.get(url)
-        r.raise_for_status()
+        r = test_client.get(url, follow_redirects=True)
+        assert r.status_code == http.client.OK
         if version == "v4":
-            products = r.json()["Info"]["Products"]
+            products = r.json["Info"]["Products"]
         elif version == "v5":
-            products = r.json()["Data"]
+            products = r.json["Data"]
         else:
             raise NotImplementedError()
         assert len(products) == 3
 
     @pytest.mark.parametrize("version", ("v4", "v5"))
-    def test_get_products_for_missing_freqmode(self, version, odinapi_service):
+    def test_get_products_for_missing_freqmode(self, version, test_client):
         url = make_dev_url(
             "{host}/rest_api/{version}/level2/{project}/{freqmode}/products/".format(
-                host=odinapi_service,
+                host="http://localhost",
                 version=version,
                 freqmode=2,
                 project=PROJECT_NAME,
             )
         )
-        r = requests.get(url)
-        r.raise_for_status()
+        r = test_client.get(url, follow_redirects=True)
+        assert r.status_code == http.client.OK
         if version == "v4":
-            products = r.json()["Info"]["Products"]
+            products = r.json["Info"]["Products"]
         elif version == "v5":
-            products = r.json()["Data"]
+            products = r.json["Data"]
         else:
             raise NotImplementedError()
         assert len(products) == 0
 
     def validate_v4_results(self, url, nr_expected, expected_l2):
-        r = requests.get(url)
-        r.raise_for_status()
-        results = r.json()["Info"]["Results"]
+        print(url)
```

Comment:
```suggestion
```
also in other places, like just below, and I may have missed a few. Maybe make a search or something?

Reply:
ok - will fix